lib/eventq/eventq_aws/aws_subscription_manager.rb

```diff
@@ -22,24 +22,28 @@ def subscribe(event_type, queue, topic_region = nil, queue_region = nil)
         topic_arn = @client.sns_helper(topic_region).public_send(method, event_type, topic_region)
         raise Exceptions::EventTypeNotFound, 'SNS topic not found, unable to subscribe' unless topic_arn
 
-        q = @manager.get_queue(queue)
-        queue_arn = @client.sqs_helper(queue_region).get_queue_arn(queue)
+        queue_arn = configure_queue(queue, queue_region)
 
-        attributes = default_queue_attributes(q, queue_arn)
-        @client.sqs(queue_region).set_queue_attributes(attributes)
+        # subscribe the queue to the topic with the namespaces provided
+        topic_namespaces.each do |namespace|
+          namespaced_topic_arn = topic_arn.gsub(":#{EventQ.namespace}-", ":#{namespace}-")
 
-        # skip subscribe if subscription for given queue/topic already exists
-        # this fixes a localstack issue: https://github.com/localstack/localstack/issues/933
-        return true if existing_subscription?(queue_arn, topic_arn)
+          # create the sns topic - this method is idempotent & returns the topic arn if it already exists
+          @client.sns.create_topic(name: "#{namespace}-#{event_type}".delete('.'))
```

Comment:
We should only execute this line if `queue.isolated`.

Reply:
looking at the code, isolated defaults to `false` and it seems that should be set to true if you dont want the topic to be automatically created (ie if it already exists)
should this line not therefore be:
```
@client.sns.create_topic(name: "#{namespace}-#{event_type}".delete('.')) unless queue.isolated
```

Reply:
Correct.